services/UserService.go

```diff
@@ -54,7 +58,11 @@ func UpdateUser(partialUpdate bool, user users.User) (*users.User, *errors.RestE
 			}
 		}
 		if user.Username != "" {
-			current.Username = user.Username
+			if err := user.UsernameValidation(); err != nil {
+				return nil, err
+			} else {
```

Comment:
Just a subtle comment: The else statement is not necessary since you will never reach to this part of the code if the request falls into the error part.

Reply:
Hm I see your point, maybe it could be refactored but if I take it out as it is, though an error displays, it is an internal service error not a bad request error like we need.